src/event.js

```diff
@@ -392,90 +422,93 @@ jQuery.event = {
 		return handlerQueue;
 	},
 
-	// Includes some event props shared by KeyEvent and MouseEvent
-	props: ( "altKey bubbles cancelable ctrlKey currentTarget detail eventPhase " +
-		"metaKey relatedTarget shiftKey target timeStamp view which" ).split( " " ),
+	// Includes all common event props including KeyEvent and MouseEvent specific props
+	props: ( "altKey bubbles cancelable ctrlKey detail eventPhase " +
+		"metaKey shiftKey view which char charCode key keyCode " +
+		"button buttons clientX clientY offsetX offsetY pageX pageY screenX screenY toElement"
+	).split( " " ),
 
 	fixHooks: {},
 
-	keyHooks: {
-		props: "char charCode key keyCode".split( " " ),
-		filter: function( event, original ) {
+	propHooks: {
+		which: function( event ) {
+			var button = event.button;
 
 			// Add which for key events
-			if ( event.which == null ) {
-				event.which = original.charCode != null ? original.charCode : original.keyCode;
+			if ( event.which == null && rkeyEvent.test( event.type ) ) {
+				return event.charCode != null ? event.charCode : event.keyCode;
 			}
 
-			return event;
-		}
-	},
+			// Add which for click: 1 === left; 2 === middle; 3 === right
+			// Note: button is not normalized, so don't use it
```

Comment:
Might as well remove this line since we pass it through and it was almost intended to be for end users.